src/olympia/users/tasks.py

```diff
@@ -112,3 +123,145 @@ def sync_blocked_emails(batch_size=BATCH_SIZE, **kw):
 
                 email_blocks = [SuppressedEmail(email=record[3]) for record in batch]
                 SuppressedEmail.objects.bulk_create(email_blocks, ignore_conflicts=True)
+
+
+@task(autoretry_for=(HTTPError, Timeout), max_retries=5, retry_backoff=True)
+def send_suppressed_email_confirmation(suppressed_email_verification_id):
+    verification = SuppressedEmailVerification.objects.filter(
+        id=suppressed_email_verification_id
+    ).first()
+
+    if not verification:
+        raise Exception(f'invalid id: {suppressed_email_verification_id}')
+
+    email = verification.suppressed_email.email
+
+    url = (
+        f'{settings.SOCKET_LABS_HOST}servers/{settings.SOCKET_LABS_SERVER_ID}/'
+        f'suppressions/remove?emailAddress={email}'
+    )
+    headers = {
+        'authorization': f'Bearer {settings.SOCKET_LABS_TOKEN}',
+    }
+
+    try:
+        task_log.info(f'removing email: {email} from suppression list')
+        response = requests.delete(url, headers=headers)
+        response.raise_for_status()
+    except HTTPError as http_err:
+        if response.status_code == 404:
+            task_log.warn(f'email not found in suppression list: {email}')
+            pass
+        else:
+            raise http_err
+
+    code_snippet = str(verification.confirmation_code)[-5:]
+
+    verification.status = SuppressedEmailVerification.STATUS_CHOICES.Pending
+
+    confirmation_link = (
+        reverse('devhub.email_verification')
+        + '?code='
+        + str(verification.confirmation_code)
+    )
+
+    send_mail_jinja(
+        gettext(f'Verify your email ({code_snippet})'),
+        'devhub/emails/verify-email-requested.ltxt',
+        {
+            'confirmation_link': confirmation_link,
+        },
+        recipient_list=[email],
+    )
+
+    verification.save()
+    check_suppressed_email_confirmation.delay(verification.id)
+
+
+@task(
+    autoretry_for=(
+        HTTPError,
+        Timeout,
+    ),
+    max_retries=5,
+    retry_backoff=True,
+)
+def check_suppressed_email_confirmation(suppressed_email_verification_id, page_size=5):
+    verification = SuppressedEmailVerification.objects.filter(
+        id=suppressed_email_verification_id
+    ).first()
+
+    if not verification:
+        raise Exception(f'invalid id: {suppressed_email_verification_id}')
+
+    email = verification.suppressed_email.email
+
+    current_count = 0
+    total = 0
+
+    code_snippet = str(verification.confirmation_code)[-5:]
+    path = f'servers/{settings.SOCKET_LABS_SERVER_ID}/reports/recipient-search/'
+
+    # socketlabs might set the queued time any time between 00:00 and 23:59
+    # so we need to check the full day before creation
+    # and to the end of the current day
+    start_date = (verification.created - datetime.timedelta(days=1)).replace(
+        hour=0, minute=0, second=0, microsecond=0
+    )
+    end_date = datetime.datetime.now().replace(
+        hour=23, minute=59, second=59, microsecond=999999
+    )
```

Comment:
is there any advantage to doing this over creating new datetimes with just the original year,month,day?

Reply:
I guess you mean something like

```python
start_date = datetime.datetime(
  year=verification.created.year,
  month=verification.created.month,
  day=verification.created.day,
  hour=0,
  minute=0,
  second=0,
  microsecond=0
)
```

I think the way it is currently is easier to read but I'll change it if you disagree.

Reply:
yeah, but you don't need the `=0` because those are the defaults.  (And I personally wouldn't bother with the end_date being the last possible microsecond - one microsecond more seems fine too?)

Reply:
I see, didn't think of this. So I've changed it to something like

```python
    before = verification.created - datetime.timedelta(days=1)
    start_date = datetime.datetime(
        year=before.year,
        month=before.month,
        day=before.day,
    )
    end_date = datetime.datetime.now() + datetime.timedelta(days=1)
```

So we get the day before creation and the day after today.